pkg/k8s/deleter/deleter.go

```diff
@@ -0,0 +1,188 @@
+	services := &corev1.ServiceList{}
+	err := c.List(ctx, services)
+	if err != nil {
+		return fmt.Errorf("error listing Services: %w", err)
+	}
+
+	svcsFailedToBeDeleted := make(objectMetaList, 0)
+	for i := range services.Items {
+		svc := &services.Items[i]
+		if needsDelete(svc) {
+			log.Info(fmt.Sprintf("Deleting Service %s/%s", svc.Namespace, svc.Name))
+			if err = c.Delete(ctx, svc); err != nil {
+				log.Error(
+					err,
+					fmt.Sprintf(
+						"Error deleting Service %s/%s",
+						svc.Namespace,
+						svc.Name,
+					),
+				)
+				svcsFailedToBeDeleted = append(svcsFailedToBeDeleted, svc.ObjectMeta)
+				continue
+			}
+			if err = waitForServiceDeletion(ctx, c, svc); err != nil {
+				log.Error(
+					err,
+					fmt.Sprintf(
+						"Error waiting for Service to be deleted %s/%s",
+						svc.Namespace,
+						svc.Name,
+					),
+				)
+				svcsFailedToBeDeleted = append(svcsFailedToBeDeleted, svc.ObjectMeta)
+			}
+		}
+	}
+	if len(svcsFailedToBeDeleted) > 0 {
+		return failedToDeleteServicesError(svcsFailedToBeDeleted)
+	}
+
+	return nil
+}
+
+// needsDelete will return true if the Service needs to be deleted to allow for cluster cleanup.
+func needsDelete(service *corev1.Service) bool {
+	if service.Spec.Type != corev1.ServiceTypeLoadBalancer ||
+		len(service.Status.LoadBalancer.Ingress) == 0 {
+		return false
+	}
+	return service.Status.LoadBalancer.Ingress[0].IP != "" ||
+		service.Status.LoadBalancer.Ingress[0].Hostname != ""
+}
+
+func waitForServiceDeletion(
+	ctx context.Context,
+	c client.Client,
+	service *corev1.Service,
+) (err error) {
+	backoff := wait.Backoff{
+		Duration: 1 * time.Second,
+		Factor:   1.5,
+		Jitter:   0,
+		Steps:    13,
+	}
+	// the error is always nil to retry but is captured in the named return err
+	_ = wait.ExponentialBackoff(backoff, func() (bool, error) {
+		key := client.ObjectKey{
+			Namespace: service.Namespace,
+			Name:      service.Name,
+		}
+		err = c.Get(ctx, key, service)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				err = nil
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+
+	return
+}
+
+func failedToDeleteServicesError(svcsFailedToBeDeleted objectMetaList) error {
+	//nolint:goerr113 // This error is specific to this function
+	return fmt.Errorf("the following Services could not be deleted "+
+		"and must cleaned up manually before deleting the cluster: %s", svcsFailedToBeDeleted.asCommaSeparatedString())
+}
+
+func ShouldDeleteServicesWithLoadBalancer(cluster *v1beta1.Cluster) (bool, error) {
+	// use the Cluster annotations to skip deleting
+	val, found := annotations.Get(cluster, constants.LoadBalancerGCAnnotation)
+	if !found {
+		val = "true"
+	}
+	shouldDeleteBasedOnAnnotation, err := strconv.ParseBool(val)
+	if err != nil {
+		return false, fmt.Errorf(
+			"converting value %s of annotation %s to bool: %w",
+			val,
+			constants.LoadBalancerGCAnnotation,
+			err,
+		)
+	}
+
+	// use the Cluster phase to determine if its safe to skip deleting
```

Comment:
nit
```suggestion
	// use the Cluster phase to determine if it's safe to skip deleting
```

Reply:
😅

Reply:
Punctuation is important :joy: i feel like a bit of a **** doing this now :facepalm:

Reply:
Haha I'm glad you did! Now its out there though so I'll never get into that writing program at Harvard 😆